### ISSUE TITLE
Move writer collect_asset_docs logic into detector

### ISSUE THREAD

**Issue by rosesyrett** [bluesky/ophyd-async#99]
Currently, the Flyer has a `collect_asset_docs` method which calls (in parallel) the `collect_stream_docs(idx)` methods of each writer underneath it. This is essentially a way to concurrently collect the asset docs of all writers underneath the flyer.

A new issue has been added, https://github.com/bluesky/bluesky/issues/1649, which will allow this logic to be moved into the Detector. So the plan is to:

1. Remove collect_asset_docs entirely from the Flyer; the Flyer should be transparent to any document generation. Remove it from the `DetectorGroupLogic` protocol and implementation(s) also for the same reason.
2. Make the StandardDetector obey the new protocol in the above issue; that is, it should have a `collect_asset_docs(idx)` method as well as something equivalent to `get_index`, which should mean it calls it's underlying writer to figure out what index of data it has written.

In order to test this, we should probably do https://github.com/bluesky/ophyd-async/issues/91 alongside this.
